### PR TITLE
Added new api EpgEventIcon for ListItem

### PR DESCRIPTION
**Pull request by lightglitch**
## Description
Added new API to use EPG program images on PVR skins

## Motivation and Context
I want my pvr to show the current program image instead of the channel icon, if exists

## How Has This Been Tested?
I was unable to test it because I was unable to compile kodi (still trying) so help is needed for this .

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [x] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by ksooo** [comment]
Thanks for your contribution.

> I want my pvr to show the current program image instead of the channel icon, if exists

Looking at the code, I doubt that your approach will work. When creating a `CFileItem ` with an `CPVREpgInfoTag`, the value returned by `CPVREpgInfoTag:Icon` will be preferred and set as icon for the icon. The channel icon is only a fallback:

https://github.com/xbmc/xbmc/blob/master/xbmc/FileItem.cpp#L151

``` c++
  if (!tag->Icon().empty())
    SetIconImage(tag->Icon());
  else if (tag->HasChannel() && !tag->Channel()->IconPath().empty())
    SetIconImage(tag->Channel()->IconPath());
```

Generally, you never should submit untested PRs.

**Review by ksooo — CHANGES_REQUESTED** [review]
This does not work.

**Comment by lightglitch** [comment]
I'm closing until I'm able to test it but the goal was to show the program image in the channels list instead of the channel icon.

**Comment by ksooo** [comment]
Reading the PR description and the code a second time I must admit that I got everything completely wrong. :-)

The code is fine as it is. Thus I will reopen the PR and in case code compiles I will merge it.

Sorry for the noise.

**Comment by ksooo** [comment]
jenkins build this please

**Comment by ksooo** [comment]
@ronie here is another new guiinfo label...
@sualfred fyi

**Review by ksooo — APPROVED** [review]

**Comment by lightglitch** [comment]
@ksooo No problem, glad I could contribute.

File: xbmc/GUIInfoManager.cpp [hunk 1]
```diff
@@ -3645,6 +3645,11 @@ const infomap container_str[]  = {{ "property",         CONTAINER_PROPERTY },
 ///                  _string_,
 ///     Returns the title of the epg event associated with the item\, if any
 ///   }
+///   \table_row3{   <b>`ListItem.EpgEventIcon`</b>,
+///                  \anchor ListItem_EpgEventIcon
+///                  _string_,
+///     Returns the thumbnail (if it exists) the epg event associated with the item\, if any
```
Comment (enen92) [inline xbmc/GUIInfoManager.cpp]: Returns the thumbnail for the epg event associated with the item (if it exists)

**Comment by sualfred** [comment]
Just for confirmation -> ListItem.EpgEventIcon is working very well.

@lightglitch 
All that is missing is PVR.EpgEventIcon to display the thumb of the currently playing channel in the video osd of the player / fullscreeninfo dialog. Maybe you could also take care of it.

**Comment by lightglitch** [comment]
@sualfred is this already available in the latest nightly? Because I'm still having issues compiling the source code.

**Comment by sualfred** [comment]
Yes.

**Pull request merged**